File: dx-api/app/consts/referral_status.go
package consts

// Referral status values.
const (
	ReferralStatusPending  = "pending"
	ReferralStatusPaid     = "paid"
	ReferralStatusRewarded = "rewarded"
)

// ReferralStatusLabels maps each referral status to its Chinese label.
var ReferralStatusLabels = map[string]string{
	ReferralStatusPending:  "待激活",
	ReferralStatusPaid:     "已付费",
	ReferralStatusRewarded: "已发放",
}

File: dx-web/src/consts/referral-status.ts
export const REFERRAL_STATUSES = {
  PENDING: "pending",
  PAID: "paid",
  REWARDED: "rewarded",
} as const;

export type ReferralStatus =
  (typeof REFERRAL_STATUSES)[keyof typeof REFERRAL_STATUSES];

export const REFERRAL_STATUS_LABELS: Record<ReferralStatus, string> = {
  pending: "待验证",
  paid: "已付费",
  rewarded: "已发放",
};
